presto-main/src/main/java/com/facebook/presto/operator/aggregation/ApproximateDoublePercentileArrayAggregations.java

```diff
@@ -0,0 +1,95 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation;
+
+import com.facebook.presto.operator.aggregation.state.DigestAndPercentileArrayState;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.block.BlockBuilderStatus;
+import com.facebook.presto.spi.type.StandardTypes;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.type.ArrayType;
+import com.facebook.presto.type.SqlType;
+import com.google.common.collect.ImmutableList;
+import io.airlift.stats.QuantileDigest;
+
+import java.util.List;
+
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.util.Failures.checkCondition;
+import static com.google.common.base.Preconditions.checkState;
+
+@AggregationFunction("approx_percentile")
+public final class ApproximateDoublePercentileArrayAggregations
+{
+    public static final InternalAggregationFunction DOUBLE_APPROXIMATE_PERCENTILE_ARRAY_AGGREGATION = new AggregationCompiler().generateAggregationFunction(ApproximateDoublePercentileArrayAggregations.class, new ArrayType(DOUBLE), ImmutableList.<Type>of(DOUBLE, new ArrayType(DOUBLE)));
+
+    private ApproximateDoublePercentileArrayAggregations() {}
+
+    @InputFunction
+    public static void arrayInput(DigestAndPercentileArrayState state, @SqlType(StandardTypes.DOUBLE) double value, @SqlType("array<double>") Block quantilesArrayBlock)
+    {
+       ApproximateLongPercentileArrayAggregations.arrayInput(state, doubleToSortableLong(value), quantilesArrayBlock);
+    }
+
+    @CombineFunction
+    public static void combine(DigestAndPercentileArrayState state, DigestAndPercentileArrayState otherState)
+    {
+        ApproximateLongPercentileArrayAggregations.combine(state, otherState);
+    }
+
+    @OutputFunction("array<double>")
+    public static void output(DigestAndPercentileArrayState state, BlockBuilder out)
+    {
+        QuantileDigest digest = state.getDigest();
+        List<Double> percentiles = state.getPercentiles();
+
+        BlockBuilder blockBuilder = DOUBLE.createBlockBuilder(new BlockBuilderStatus(), percentiles.size());
+        for (int i = 0; i < percentiles.size(); i++) {
+            Double percentile = percentiles.get(i);
+            checkState(percentile != -1.0, "Percentile is missing");
+            checkCondition(0 <= percentile && percentile <= 1, INVALID_FUNCTION_ARGUMENT, "Percentile must be between 0 and 1");
+            try {
+                DOUBLE.writeDouble(blockBuilder, longToDouble(digest.getQuantile(percentile)));
+            }
+            catch (NullPointerException e) {
```

Comment:
Remove `try ... catch`. It's almost never a good idea to catch `NullPointerException` and throw it away.